nifi-nar-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -0,0 +1,432 @@
+                    }
+                }
+
+                route(flowfile, responseFlowFile, session, context, httpResponseEntity.statusCode());
+            }
+        } catch (Exception e) {
+            if (flowfile == null) {
+                logger.error("Request Processing failed", e);
+                context.yield();
+            } else {
+                logger.error("Request Processing failed: {}", flowfile, e);
+                session.penalize(flowfile);
+                flowfile = session.putAttribute(flowfile, GET_WORKDAY_REPORT_JAVA_EXCEPTION_CLASS, e.getClass().getName());
+                flowfile = session.putAttribute(flowfile, GET_WORKDAY_REPORT_JAVA_EXCEPTION_MESSAGE, e.getMessage());
+                session.transfer(flowfile, FAILURE);
+            }
+
+            if (responseFlowFile != null) {
+                session.remove(responseFlowFile);
+            }
+        }
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        List<ValidationResult> results = new ArrayList<>(super.customValidate(validationContext));
+        if (validationContext.getProperty(FIELDS_TO_HASH).isSet() && !validationContext.getProperty(RECORD_READER_FACTORY).isSet()) {
+            results.add(new ValidationResult.Builder()
+                .valid(false)
+                .explanation("Record-Reader and Record-Writer must be set if you would like to hash specific fields")
+                .subject("Workday report configuration")
+                .build());
+        }
+        return results;
+    }
+
+    /*
+     *  If we have no FlowFile, and all incoming connections are self-loops then we can continue on.
+     *  However, if we have no FlowFile and we have connections coming from other Processors, then
+     *  we know that we should run only if we have a FlowFile.
+     */
+    private boolean skipExecution(ProcessContext context, FlowFile flowfile) {
+        return context.hasIncomingConnection() && flowfile == null && context.hasNonLoopConnection();
+    }
+
+    private FlowFile createResponseFlowFile(FlowFile flowfile, ProcessSession session, ProcessContext context, HttpResponseEntity httpResponseEntity)
+        throws IOException, SchemaNotFoundException, MalformedRecordException {
+        FlowFile responseFlowFile = null;
+        String hashingAlgorithm = context.getProperty(HASHING_ALGORITHM).getValue();
+        Set<String> columnsToHash = Optional.ofNullable(context.getProperty(FIELDS_TO_HASH).evaluateAttributeExpressions(flowfile).getValue()).map(String::trim)
+            .map(columns -> columns.split(COLUMNS_TO_HASH_SEPARATOR)).map(Arrays::stream).map(columns -> columns.collect(Collectors.toSet())).orElse(Collections.emptySet());
+        try {
+            if (isSuccess(httpResponseEntity.statusCode())) {
+                responseFlowFile = flowfile != null ? session.create(flowfile) : session.create();
```

Comment:
Recommend adjusting the logic as follows:
```suggestion
                responseFlowFile = flowfile == null ? session.create() : session.create(flowfile);
```